### PR TITLE
debug: tracing spans for ShardsManagerActor

### PR DESCRIPTION
**Pull request by jakmeier**
Adds spans for `ShardsManagerRequestFromClient` and `ShardsManagerRequestFromNetwork`. Also tracks it with parents spans across actor boundary.

This allows tracing collectors to display to show meaningful traces for the ShardsManagerActor. For example, in Jaeger, I can see where these requests come from, how long they take to execute, and what type of message they are tracking.

### CODE REVIEW

**Comment by jakmeier** [comment]
This extra observability was useful for me recently when working on some prototypes. I think it makes sense to commit it to maser, too.

But it's a bit messier then I hoped it would be. @robin-near  Please let me know if there is a better approach to make `WithSpanContext` work with the testing setup.

Alternatively, I can also remove the `WithSpanContext` shenanigans and only add the simple traces with `"type=..." info, which is quite useful on its own. Let me know what you prefer.

**Review by nikurt — DISMISSED** [review]

**Review by robin-near — CHANGES_REQUESTED** [review]
Wait, I wish you knew this... would've saved you a lot of work here. When you have an `Addr<Y>` where Y can handle `WithSpanContext<X>`, you can convert it to a `Sender<X>` by just doing .with_auto_span_context(). This is how the Sender framework avoids having to use .with_span_context() all over the place.

**Comment by jakmeier** [comment]
@robin-near Thank for this info, I suspected you have a better solution than my horrible code, hence why I requested a review from you in the first place :)

I was able to clean up most of the code. But struggling a bit with `fake_client::Fake` used in tests. Is there a good way to make it work? (See compilation errors in CI)

**Comment by robin-near** [comment]
I think you simply need to avoid span context everywhere except when converting actix to Sender. For Fake there's also no need to involve it. See https://github.com/near/nearcore/commit/23a83dd751e9a096633feb0e16f249aad33a21cf ; this compiles for me.

**Comment by jakmeier** [comment]
Oh, right of course. I completely forgot I made changes to the Fake client, just needed to undo those :facepalm: Sorry, I should have looked at the final diff again, would have been obvious :D

But I think now it should be all ready for final review now @robin-near

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -1961,6 +1967,12 @@ impl ShardsManager {
     }
 
     pub fn handle_network_request(&mut self, request: ShardsManagerRequestFromNetwork) {
+        let _span = tracing::debug_span!(
+            target: "chunks",
+            "shards_manager_request_from_network",
+            "type" = <&ShardsManagerRequestFromNetwork as Into<&'static str>>::into(&request)
```
Comment (robin-near) [inline chain/chunks/src/lib.rs:1973]: nit: I think both of these work?

`Into::<&'static str>::into(&request)`
`<&'static str>::from(&request)`

**Review by robin-near — APPROVED** [review]

**Comment by robin-near** [comment]
Thank you @jakmeier !!

File: chain/chunks/src/lib.rs (hunk 1 above)
Comment (jakmeier) [reply chain/chunks/src/lib.rs]: Ah yes, that's much nicer, thanks.

**Comment by jakmeier** [comment]
Pushing the fix for the nit made the approval stale -> please approve again @robin-near

**Review by robin-near — APPROVED** [review]

**Comment by robin-near** [comment]
Huh, that's so weird...

**Comment by jakmeier** [comment]
I think it's working as intended: https://near.zulipchat.com/#narrow/stream/295302-general/topic/Github.20nearcore.20branch.20protection

**Pull request merged**